subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/VariantResolveMetadata.java

```diff
@@ -32,4 +33,6 @@
     AttributeContainerInternal getAttributes();
 
     List<? extends ComponentArtifactMetadata> getArtifacts();
+
+    CapabilitiesMetadata getCapabilitiesMetadata();
```

Comment:
Nit: for consistency, should be `getCapabilities()`

Reply:
On another note, do `VariantResolveMetadata` actually _need_ to expose the capabilities? I only ask because we don't expose dependencies or constraints here.

Reply:
It's a tough call. I wondered too, and tried _without_ defining `getCapabilities()` here. But the consequence was "duplicate modeling" due to the fact that `VariantResolveMetadata` represent _where we want to go_, vs `ConfigurationMetadata`, which represents the old stuff. Since this is all new, I chose not to duplicate things and expose it directly. We can probably avoid it, but it comes with additional complexity that I feel is not needed.